com.unity.netcode.gameobjects/Components/NetworkTransform.cs

```diff
@@ -182,37 +182,16 @@ internal bool IsTeleportingNextFrame
             internal float ScaleX, ScaleY, ScaleZ;
             internal double SentTime;
 
-            internal Vector3 Position
-            {
-                get { return new Vector3(PositionX, PositionY, PositionZ); }
-                set
-                {
-                    PositionX = value.x;
-                    PositionY = value.y;
-                    PositionZ = value.z;
-                }
-            }
-
-            internal Vector3 Rotation
-            {
-                get { return new Vector3(RotAngleX, RotAngleY, RotAngleZ); }
-                set
-                {
-                    RotAngleX = value.x;
-                    RotAngleY = value.y;
-                    RotAngleZ = value.z;
-                }
-            }
+            internal bool IsDirty;
 
-            internal Vector3 Scale
+            /// <summary>
+            /// This will reset the NetworkTransform BitSet
+            /// </summary>
+            internal void ClearBitSetForNextTick()
             {
-                get { return new Vector3(ScaleX, ScaleY, ScaleZ); }
-                set
-                {
-                    ScaleX = value.x;
-                    ScaleY = value.y;
-                    ScaleZ = value.z;
-                }
+                var localSpace = (ushort)(m_Bitset & (1 << k_InLocalSpaceBit));
```

Comment:
bitset is for dirtying certain axis and fields in the transform's state. Why are we keeping localSpace always dirty?

Reply:
We are preserving local space settings there.
bitset axial values are now reset every tick, with the exception of the local space setting which applies for all ticks until it is changed by authority.

Reply:
why are we keeping local space settings for all ticks?

Reply:
Because that is only changed by authority on the authority side.  Consider the following:

- First 100 ticks it is in localspace
  - We want to keep that setting in the state
- Tick 101 authority changes it to globalspace
  - We want to keep that setting in the state too.

It just preserves localspace if it is set...that is all.

Reply:
oh wait, I think I forgot. All others bits are for dirtying, this bit is the actual value, is it? (might want to add a quick comment on this, I forgot about this after almost a year not touching this code base)

Reply:
Yep!
The rest are ok to reset when preparing for the next tick, but we have to preserve the local space flag as that is only changed when the NetworkTransform.InLocalSpace is changed on the authoritative side.